src/coap_gnutls.c

```diff
@@ -1211,16 +1238,19 @@ do_gnutls_handshake(coap_session_t *c_session, coap_gnutls_env_t *g_env) {
     G_ACTION(gnutls_alert_send(g_env->g_session, GNUTLS_AL_FATAL,
                                                  GNUTLS_A_DECRYPT_ERROR));
     c_session->dtls_event = COAP_EVENT_DTLS_CLOSED;
+    ret = -1;
     break;
   case GNUTLS_E_UNKNOWN_CIPHER_SUITE:
   case GNUTLS_E_TIMEDOUT:
```

Comment:
I keep wondering why the compiler does not emit a warning for this. I suggest adding a `/* fall through */` for clarity.

Reply:
It is more about returning something other than -1 on error is the issue here.  Updated as suggested.